recipes/libcurl/all/conanfile.py

```diff
@@ -246,6 +246,9 @@ def _get_configure_command_args(self):
 
         params.append("--with-brotli" if self.options.with_brotli else "--without-brotli")
 
+        # TODO: prevent linking of system-installed zstd
+        params.append("--without-zstd")
```

Comment:
of course this is a stop-gap workaround until someone finds the time to enable zstd in the recipe properly (zstd was introduces with libcurl 7.72, so the options would have to be done based on the version)

Reply:
Did you check that this option is supported by all libcurl versions supported in CCI?

Reply:
Does this work?
1. Add `with_zstd` to the `options` and `default_options` dict
2. Remove `with_zstd` from `options` in `config_options` if `tools.Version(self.version) < 7.72.0`
3. Add a `_with_zstd` property method that does the following:
    ```python
    @property
    def _with_zstd(self):
        return self.options.get_safe("with_zstd", False)
    ```
4. Add the following argument to the configure script:
`params.append("--with-zstd" if self._with_zstd else "--without-zstd")`

Reply:
I don't think it works, `--with-zstd` doesn't exist in `configure.ac` before `7.72.0`

Reply:
Thanks @SpaceIm , indeed older versions will fail when encountering an unknown option.
Then the steps should be:
1. Add with_zstd to the options and default_options dict
2. Remove with_zstd from options in config_options if tools.Version(self.version) < 7.72.0
3. Add the arguments conditionally to the configure script.
    ```python
    if tools.Version(self.version) >= "7.72.0":
        params.append("--with-zstd" if self.options.with_zstd else "--without-zstd")
    ```

Reply:
@timblechmann doesn't want to support zstd in this PR (it requires additional work, in `requirements` method, but also in CMake build and `package_info`), just avoid silent link of system zstd during build for autotools builds.

Reply:
Aight! I see!
How about adding a FIXME about missing zstd for versions >= 7.72.0?